**Problem Statement 1:**

Blood glucose levels for obese patients have a mean of 100 with a standard deviation of 15. A researcher thinks that a diet high in raw cornstarch will have a positive effect on blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an effect or not.

In [1]:
mean = 100
std = 15
sample = 36
sample_mean  = 108

std_error = std/sample**0.5
z = (sample_mean-mean)/(std_error)
print("Standard error %0.2f"%std_error)
print("Z_score %0.2f"%z)

Standard error 2.50
Z_score 3.20


By looking at z- table and p-value associated with 3.20 is 0.9993. 

The probability of having value less than 108 is 0.9993 and more than or equals to 108 is (1-0.9993)=0.0007.

Since the probability of having mean glucose level more than or equals to 108  is 0.0007 which is  less than 0.05  so we will reject the Null hypothesis i.e. there is raw cornstarch effect.

**Problem Statement 2:**

In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple random sample of 100 voters are surveyed from each state. What is the probability that the survey will show a greater percentage of Republican voters in the second state than in the first state?

In [2]:
#State1
republic1 = .52
democrats1 = 0.48
n1 = 100

#State2
republic2 = 0.47
democrats2 = 0.53
n2 = 100

mu = republic1-republic2
print(f"Mean Difference in number of  republican  in both states: {mu}")


std = ((republic1 * democrats1 ) / n1 + (republic2 * democrats2) /n2)**0.5
print(f"std: {std}")

'''Difference should be less than 0 for greater percentage of Republican voters 
in the second  state than in the first state'''
x = 0
z_score = (x-mu)/std
print(f"z score : {z_score}")

Mean Difference in number of  republican  in both states: 0.050000000000000044
std: 0.07061869440877536
z score : -0.7080278164104213


From Z score table -0.708 refers to 0.24

Hence 0.24 is the probability that the survey will show a greater percentage of Republican voters in the second  state than in the first state

**Problem Statement 3:**

You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard deviation is 209. How well did you score on the test compared to the average test taker?

In [3]:
mean = 1026
std = 209
z =  (1100-mean)/std

print("The z value is {0}. This means that your score was {0} std deviaton above the mean.".format(z))

The z value is 0.35406698564593303. This means that your score was 0.35406698564593303 std deviaton above the mean.
